CI broken: Azure.Security.KeyVault.Certificates 4.11.0 not published on nuget.org

## CI broken on main: .NET 10 SDK is not installed by the workflow

All `ci (app: …) / Build, Test, Analyze` jobs are failing with:

```
error NU1102: Unable to find package Azure.Security.KeyVault.Certificates with version (>= 4.11.0)
  - Found 35 version(s) in nuget.org [ Nearest version: 4.9.0-beta.1 ]
```

The failure is on every project that pulls in the Azure SDK chain (Authorization, AccessManagement, integration libs, the test projects). It hits **main itself** — confirmed on the most recent main run [25667949801](https://github.com/Altinn/altinn-authorization-tmp/actions/runs/25667949801) (commit 4c822086) — and every PR branched off post-#3066, including this PR's CI for #3101 [run 25670644283](https://github.com/Altinn/altinn-authorization-tmp/actions/runs/25670644283).

## Root cause

PR #3066 ("dotnet 10 upgrade") bumped:

- `global.json` SDK to `10.0.107`
- `src/Directory.Build.props` `<TargetFramework>` to `net10.0`
- `src/pkgs/Directory.Build.props` `<TargetFrameworks>` to `net8.0;net10.0`
- 11 test csproj `<TargetFrameworks>` from `net9.0` to `net10.0`
- Microsoft / EF Core / Npgsql packages to `10.0.x`

But [`.github/workflows/tpl-vertical-ci.yml`](https://github.com/Altinn/altinn-authorization-tmp/blob/main/.github/workflows/tpl-vertical-ci.yml#L74-L79) still pins the install to 8 and 9:

```yaml
- name: Install .NET
  uses: actions/setup-dotnet@…
  with:
    dotnet-version: |
      8.0.x
      9.0.x
```

Without the .NET 10 SDK, `dotnet workload restore` and the subsequent `dotnet build` resolve the net10 package graph against the .NET 9 SDK as a fallback. That fallback resolves the Azure SDK transitive chain (`Azure.Identity 1.17.0` → other Azure.Security.KeyVault.* refs → `Azure.Security.KeyVault.Certificates`) along a code path that floors at `>= 4.11.0` — a version that doesn't yet exist on nuget.org (latest is `4.9.0-beta.1`). The centrally-pinned `4.8.0` in `src/Directory.Packages.props` is ignored because the floor is transitive.

The intermittent main success ([run 25662581869](https://github.com/Altinn/altinn-authorization-tmp/actions/runs/25662581869) on 09469cca passed) is most likely NuGet feed cache variance — when a cached restore graph still satisfies the constraint, the build proceeds; when the cache rolls forward to the broken transitive plan, it fails.

## Suggested fix

Single-line change to `tpl-vertical-ci.yml`:

```yaml
- name: Install .NET
  uses: actions/setup-dotnet@…
  with:
    dotnet-version: |
      8.0.x
      9.0.x
      10.0.x
```

If the same `setup-dotnet` invocation exists in `cd-apps.yml` / `cd-pkgs.yml` / `cd-infra.yml` / other vertical templates, apply there too. With the .NET 10 SDK installed, NuGet should resolve the Azure SDK chain to the published `Azure.Security.KeyVault.Certificates 4.8.0` (the central pin), which satisfies the actual net10 transitive constraints.

## Verification

After the workflow update, the next CI run on main and on any open PR should restore cleanly. The Sonar / coverage steps downstream of build should then run as before — none of them care which SDK installed the workload.

## Surfaced via

PR #3101 (S1481 unused-locals sweep + bug fixes) — its initial CI run after the SonarCloud follow-up commit started failing here. The PR's code changes don't touch any Azure package or any `*.csproj` that references one; same break hits every PR branched off main after #3066 merged.

--- a/src/Directory.Packages.props
+++ b/src/Directory.Packages.props
@@ -35,7 +35,7 @@
     <PackageVersion Include="Azure.ResourceManager.PostgreSql" Version="1.3.1" />
     <PackageVersion Include="Azure.ResourceManager.ServiceBus" Version="1.1.0" />
     <PackageVersion Include="Azure.Security.KeyVault.Certificates" Version="4.8.0" />
-    <PackageVersion Include="Azure.Security.KeyVault.Secrets" Version="4.8.0" />
+    <PackageVersion Include="Azure.Security.KeyVault.Secrets" Version="4.11.0" />
     <PackageVersion Include="Azure.Storage.Blobs" Version="12.26.0" />
     <PackageVersion Include="Azure.Storage.Queues" Version="12.24.0" />
     <PackageVersion Include="Blazored.LocalStorage" Version="4.5.0" />

--- a/src/Directory.Packages.props
+++ b/src/Directory.Packages.props
@@ -34,8 +34,8 @@
     <PackageVersion Include="Azure.ResourceManager.KeyVault" Version="1.3.3" />
     <PackageVersion Include="Azure.ResourceManager.PostgreSql" Version="1.3.1" />
     <PackageVersion Include="Azure.ResourceManager.ServiceBus" Version="1.1.0" />
-    <PackageVersion Include="Azure.Security.KeyVault.Certificates" Version="4.11.0" />
-    <PackageVersion Include="Azure.Security.KeyVault.Secrets" Version="4.11.0" />
+    <PackageVersion Include="Azure.Security.KeyVault.Certificates" Version="4.8.0" />
+    <PackageVersion Include="Azure.Security.KeyVault.Secrets" Version="4.8.0" />
     <PackageVersion Include="Azure.Storage.Blobs" Version="12.26.0" />
     <PackageVersion Include="Azure.Storage.Queues" Version="12.24.0" />
     <PackageVersion Include="Blazored.LocalStorage" Version="4.5.0" />

